### DISCUSSION TITLE
Help with gridding 1D case

### DISCUSSION THREAD

**Discussion by schmidtijoe** [mmuckley/torchkbnufft#93] [General]
Hello,
Thanks for the nice package!
I have a bit of a different use-case. I got a sequence with various non-uniform readout line samplings but matching phase encodes. If i suppose the phase encode errors are negligible, i basically just want to grid all of my read lines in 1D. Since i need another recon method afterwards i was only looking at the `KbInterpAdjoint` method of the interpolation module.
- I got N batches and C coils, (data dims [N, C, k-length), trajectory dims [N, 1, k-length]).
- set the grid and image dimension to 1d tuples
- set the k-space trajectory per batch to radians / voxel.

The interpolation seems to work, however, it appears that the output pushed the k-space center towards the edges and i appear to get edge artifacts for a partial fourier region which is not acquired (i guess this is acceptable since using interpolation to fill the non acquired data with 0s is not the intended use). When i am simply using a torch.fft.ifftshift it seems to do the job, i was just wondering if i am missing something or should fftshift the input prior to application of the interpolation. Since its not mentioned in its respective doc.

Any hints are welcome !
All the best!

**Comment by mmuckley** [comment]
Hello @schmidtijoe, the fftshift is normally built into the phase of the output data from the interpolation operation:

https://github.com/mmuckley/torchkbnufft/blob/20c45a23b4d2894fd118904131129d51104cd64f/torchkbnufft/_nufft/interp.py#L679-L686

If it helps, you can see how the interpolation operation is wrapped with the inverse FFT here:

https://github.com/mmuckley/torchkbnufft/blob/20c45a23b4d2894fd118904131129d51104cd64f/torchkbnufft/functional/nufft.py#L194-L260

So once you have your coordinates and apply interpolation, the relationship between the image and the data should be a simple FFT.

Let me know if this helps. Also might be helpful to see an image of what you mean by "pushed the k-space center towards the edges" to help any further debugging.

**Reply by schmidtijoe** [reply]
Yeah thanks, with that phrase i meant exactly that it looks like an fft shift applied along my 1 dimensional data. Which was also what i did to  "mitigate" the problem. I was just wondering if i did something wrong since it wasn't obvious to me that the output should be related via the shift. Thanks for the clarification!
